### ISSUE TITLE
Wireshark+Gnuplots conflicting with Vim (dyld-link breaks on version difference)

### ISSUE THREAD

**Issue by dougpagani** [Homebrew/brew#3593]
CHECKLIST:

- [x] ran `brew update`
- [x] ran `brew doctor`, attached as a file `brew_doctor.txt`
[brew_doctor.txt](https://github.com/Homebrew/brew/files/1580854/brew_doctor.txt)
- [x] ran `brew config`, attached as a file `brew_config.txt`
[brew_config.txt](https://github.com/Homebrew/brew/files/1580850/brew_config.txt)
----------------------

**Trying to do:** Get all packages to work on the same system.
**What happened:** 

> dyld: Library not loaded: /usr/local/opt/lua/lib/liblua.5.2.dylib
>   Referenced from: /usr/local/bin/vim
>   Reason: image not found
> Abort trap: 6

---------------------------------------------

I have seen some discussion on this between the maintainers, in dreading the lua-version-migration and its messy consequences, though admittedly when it comes to package-management it mostly went over my head. So, I realize you are probably aware of this problem, and it may be "not a brew problem." But in the case that there is a viable solution you have, or that this behavior is a brew-issue...

Vim 8.x uses the old Lua. So, when its executable paths-out for the 5.2 dylib, if the /usr/local/opt/lua symlink is pointing to the 5.3 version in the Cellar, it will break when it doesn't find its 5.2. I do get that. 

I tried reinstalling `brew vim --with-lua`, and it does recompile. But it then stomps-over the symlink for dylib 5.3, **breaking wireshark & gnuplots, which rely on 5.2**. This was a problem I've just been dealing with by invoking `brew switch lua 5.3.4_2` / `5.2.4_4` every-time I hit the tedious dyld-error. However, there are contexts where I'd like to use Wireshark+Vim, and Gnuplots+Vim. This results in ~50 switches per-session, which distracting fast. 

I may be doing this incorrectly, or have not sufficiently reviewed the past-issues; I did, and could not find anything. Maybe Brew just doesn't do multiple-version dependency management, or perhaps it does and I've hit a weird problem with my personal userflow. Or maybe Vim's tap is being managed incorrectly, and can avoid the Symlink-Stomping. 

So, there are perhaps two sub-issues gnarled in this.

1. I'd like to be able to use packages which depend on two different lua versions, without having to switch back and forth in a single session, or wrap scripts with bash in order to do `brew switch lua ...`. Is there a Brew way to do this, and if not, is there some kind of stop-gap besides having to run brew-swtich-lua each time I hit a dyld error? I was thinking I could just manually modify the executable which references the opt of lua, by digging through the hex for the reference, and pointing it directly to the Cellar-Version. This may not work, since _I think_ there may be multiple links from the Dependent executables, looking to many components of lua, not just the *.dylib.

2. It took me about 6 hours to figure out what was happening, since I would cyclically reinstall the packages, not realizing I was breaking others in the process. I don't know if it is the formula's responsibility to inform you it's stomped the symlink that other brew-packages are dynamically linked to, but if Brew is supposed to, consider this an instance of a bug for that functionality.

I'm probably making my life unnecessarily difficult, and this may be a *me* issue, not a *brew-issue*, so if that is the case, please let me know where my incompetence lies.

**Comment by MikeMcQuaid** [comment]
> [x] ran brew config, but I think the information there is probably irrelevant, and at-worst distracting

From the issue template:

> **Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you do not read fill out the issue template repeatedly you will be blocked from ever submitting issues to Homebrew again.**

If you add the necessary information this issue may be reopened.

**Comment by dougpagani** [comment]
The Information is attached as two separate files: (1) `brew_doctor.txt`, (2) `brew_config.txt`

**Comment by JCount** [comment]
@dougpagani Does running `brew update && brew outdated` show anything?

**Comment by dougpagani** [comment]
> total 0
drwxr-xr-x 12 dougpa 408 Dec  7 16:58 5.2.2_1

> total 0
drwxr-xr-x 12 dougpa 408 Dec 17 04:23 2.4.3_1

**Comment by dougpagani** [comment]
> [2] [T-1] > ls /usr/local/Cellar/wireshark/
total 0
drwxr-xr-x 12 dougpa 408 Dec 17 04:23 2.4.3_1

> [3] [T-1] > ls /usr/local/Cellar/gnuplot/
total 0
drwxr-xr-x 12 dougpa 408 Dec  7 16:58 5.2.2_1

**Comment by dougpagani** [comment]
I figured out a nice workaround. For the same functionality, `vim --with-lua` may be replaced with `vim --with-luajit`, only then the lua-symlinks (of vim & fshark/gnuplot) don't clobber one another. 

This doesn't really address the underlying issues, but it does avoid the problem entirely. I figured I would post here for those googling around for a similar workaround, even if you'd like to keep the issue open for your ops.

Thanks,
-- Doug
